pkg/controller/cluster/subnet-register.go

```diff
@@ -0,0 +1,81 @@
+// This file is part of MinIO Operator
+// Copyright (c) 2022 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cluster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/minio/madmin-go"
+	miniov2 "github.com/minio/operator/pkg/apis/minio.min.io/v2"
+	"github.com/minio/operator/subnet"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (c *Controller) registerTenantInSubnet(ctx context.Context, tenant *miniov2.Tenant, adminClnt *madmin.AdminClient) error {
```

Comment:
you should mark some status field to prevent this from attempting to register the cluster every sync loop